operator/helper/multiline.go

```diff
@@ -187,3 +195,10 @@ func encodedCarriageReturn(encoding encoding.Encoding) ([]byte, error) {
 	nDst, _, err := encoding.NewEncoder().Transform(out, []byte{'\r'}, true)
 	return out[:nDst], err
 }
+
+func trimWhitespaces(data []byte) []byte {
+	// TrimLeft to strip EOF whitespaces in case of using $ in regex
+	// For some reason newline and carriage return are being moved to beginning of next log
+	// TrimRight to strip all whitespaces from the end of log
```

Comment:
Do the test cases cover all of these behaviors? If not, should we have a few more? It seems like these nuances would be easy to miss if someone makes a future change.

Reply:
Yes, if you look at the test changes, both of the cases are tested. Should I write dedicated tests for them, so it would be clear that we are explicitly testing it?

Reply:
I misunderstood when I looked earlier. I think the current structure is good.

Sorry, but one more ask - Can you add a test or two where there are multiple returns in between the start/end pattern?

Reply:
Sure, please see new commit